### PR TITLE
Added some - a shared_ptr wrapper

### PR DESCRIPTION
**Pull request by lisitsyn**
I'm back ;)

### CODE REVIEW

**Comment by iglesias** [comment]
Travis didn't like some.h. Is it compiling with C++11 enabled?

**Comment by lambday** [comment]
Yep, variadic templates :P 
BTW @lisitsyn love to see you're back :D Let's get `Some` thing going.

**Comment by lisitsyn** [comment]
Hmm sorry didn't notice we still support C++ 98 :)

Can we get rid of it? Any modern compiler supports C++11 now

**Comment by iglesias** [comment]
I think they idea is that we have C++11 as optional dependency. Maybe some
should be guarded with HAVE_CXX11 as in the other places where c++11 stuff
are used.

**Comment by lisitsyn** [comment]
@iglesias done ;)

**Comment by iglesias** [comment]
The Isomap test failed in one of the Travis jobs. Isn't that ironic? xD

**Comment by lisitsyn** [comment]
Oh that's very ironic indeed

**Comment by lisitsyn** [comment]
Other comments anyone? ;)

File: src/shogun/base/some.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+#ifdef HAVE_CXX11
+#include <memory>
+
+#include <shogun/base/SGObject.h>
+
+namespace shogun
+{
+
+    /** Shogun synonym for the std::shared_ptr
```
Comment (karlnapf) [inline src/shogun/base/some.h]: Some more explanation here? :)

**Comment by karlnapf** [comment]
Hey @lisitsyn 
Awesome! Looking forward to see this more developed and getting rid of all those ugly SG_REFs in our code!

**Comment by iglesias** [comment]
Let's merge it, then! @lisitsyn, feel free once you think it is ready -- I'm not doing it now in case you want to do something about Heiko's question above.

**Comment by vigsterkr** [comment]
some? :)

**Comment by lisitsyn** [comment]
Gonna update doc in a minute

**Comment by lisitsyn** [comment]
@vigsterkr yeah I was psychic about short and meaningful name - correct me if it is not the thing :)

**Comment by lisitsyn** [comment]
Haha sorry I made a bad rebase

**Comment by besser82** [comment]
and besser82 did fix that mess :P

**Comment by lisitsyn** [comment]
Thanks @besser82 ;)

**Comment by lisitsyn** [comment]
Ok I rethinked this thing and going to push non C++11 version soon

**Pull request merged**